### DISCUSSION TITLE
Access react context from zustand

### DISCUSSION THREAD

**Discussion by gsouf** [pmndrs/zustand#1212] [Q&A]
Hi,

My react app is using context for dependency injection. I'm using zustand for state management and I would like to access object stored in react context from a zustand store function. Is there a way to get zustand to read react context?

Example

```js
import create from 'zustand'

const useStore = create((set) => ({
  doSomething: () => {
    const service =  getServiceFromReactContext(); 
  },
}))
```

Thanks

**Comment by dai-shi** [comment]
Your zustand store is defined at module, which is an external store for React.
To somehow link between the external store and React props/contexts/useState, you need to use `useEffect` to synchronize them.

I don't think this is an answer you are looking for. Maybe the answer is "not possible". Probably, you want to inject the value to the store from the place where you created. Sorry for vague comment.

**Reply by gsouf** [reply]
@dai-shi thanks for the answer. Indeed it's not ideal. If there is no such integration between zustand and react context, then I'am wondering whether I am the only one having this kind of need? Does that mean I'm trying to do something wrong with zustand?

**Reply by dai-shi** [reply]
The point is zustand store is global, but react context is not, it's contextual. So, it's not one-to-one relationship. It's theoretically impossible to link two different things.

**Reply by gsouf** [reply]
@dai-shi but my quesiton is rather is it ok to use things like local storage in zustand?  Or doing http data-fetching?

**Reply by gsouf** [reply]
Also I'm thinking to what you're saying. In a respectuful way, I think that your assertion is wrong when you say that it is theoretically impossible. For example we can pass arguments to zustand handlers and that makes it contextual at runtime. Passing a context is like passing arguments but in a different way

**Reply by dai-shi** [reply]
> is it ok to use things like local storage in zustand? Or doing http data-fetching?

sure.

> your assertion is wrong when you say that it is theoretically impossible.

I mean if we try to "sync" two state from different worlds. It's fine to set from props/context/state with useEffect, so it this sense, it's possible. Sorry for the confusion.

**Reply by gsouf** [reply]
@dai-shi thanks for the clarifications. I'll try to make something viable and let you know what I ended up with
